app/services/process_parser.rb

```diff
@@ -3,27 +3,65 @@
 ##
 # Parsing Process creation request
 class ProcessParser
-  attr_reader :process
+  attr_reader :process_element
+
+  PROCESS_ATTRIBUTES = %i[
+    process
+    status
+    lane_id
+    lifecycle
+    error_msg
+    error_txt
+    note
+    elapsed
+  ].freeze
 
   ##
-  # @param [Nokogiri::XML::Element] process
-  def initialize(process)
-    @process = process
+  # @param [Nokogiri::XML::Element] process_element
+  def initialize(process_element)
+    @process_element = process_element
+  end
+
+  def to_h
+    hash = {}
+    PROCESS_ATTRIBUTES.each do |attribute|
+      value = public_send(attribute)
+      next if value.nil?
+
+      hash[attribute] = value
+    end
+    hash
   end
 
-  def name
-    process.attr('name')
+  def process
```

Comment:
Renamed this method in order to align the names of methods in this class with the names of fields in the database. This is what allows us to call `step.update(parser.to_h)` in the workflows controller.